core/templates/services/messenger.service.ts

```diff
@@ -60,23 +65,29 @@ interface GetPayloadType {
   stateTransition(data: StateTransitionData): StateTransitionData;
   explorationCompleted(
     data: ExplorationCompletedData): ExplorationCompletedData;
-  explorationReset(data: string): {stateName: string};
+  explorationReset(data: string): ExplorationResetData;
 }
 
+type PayloadType = (
+  HeightChangeData |
+  ExplorationCompletedData |
+  StateTransitionData |
+  ExplorationResetData
+);
+
+type HashDict = {
+  version: string | null,
+  secret: string | null,
+  tagid: string | null
```

Comment:
Please add a comment explaining why these can be null

Reply:
done